### ISSUE TITLE
DO rollback does not properly complete

### ISSUE THREAD

**Issue by jgruber** [F5Networks/f5-declarative-onboarding#36]
When removing a SelfIP because of declaration or rollback, the use of the SelfIP for other things dependence, also delcared, must be checked.

The following declaration:

```
{
    "async": true,
    "schemaVersion": "1.0.0",
    "label": "Cloudinit Onboarding",
    "class": "Device",
    "Common": {
        "poolLicense": {
            "licensePool": "BIGIPVEREGKEYS",
            "licenseType": "licensePool",
            "bigIqHost": "192.168.245.114",
            "bigIqUsername": "admin",
            "bigIpPassword": "f5c0nfig",
            "bigIpUsername": "admin",
            "bigIqPassword": "admin",
            "reachable": true,
            "class": "License"
        },
        "ntpServers": {
            "class": "NTP",
            "servers": [
                "0.pool.ntp.org",
                "1.pool.ntp.org",
                "2.pool.ntp.org"
            ]
        },
        "HA-self": {
            "trafficGroup": "traffic-group-local-only",
            "vlan": "HA",
            "allowService": "all",
            "class": "SelfIp",
            "address": "1.1.1.103/24"
        },
        "HA": {
            "interfaces": [
                {
                    "tagged": false,
                    "name": 1.1
                }
            ],
            "class": "VLAN",
            "mtu": 1450
        },
        "configsync": {
            "configsyncIp": "/Common/HA-self/address",
            "class": "ConfigSync"
        },
        "activeStandbyGroup": {
            "saveOnAutoSync": false,
            "autoSync": true,
            "fullLoadOnSync": false,
            "members": [
                "primary-waf-instance.local",
                "secondary-waf-instance.local"
            ],
            "asmSync": false,
            "type": "sync-failover",
            "class": "DeviceGroup"
        },
        "hostname": "secondary-waf-instance.local",
        "dnsServers": {
            "nameServers": [
                "8.8.8.8"
            ],
            "search": [
                "example.openstack.com"
            ],
            "class": "DNS"
        },
        "provisioningLevels": {
            "ltm": "nominal",
            "class": "Provision",
            "asm": "nominal"
        },
        "default": {
            "gw": "192.168.80.1",
            "class": "Route",
            "network": "default",
            "mtu": 1450
        },
        "dbvars": {
            "ui.advisory.enabled": true,
            "ui.advisory.text": "This device is under centralized management.",
            "class": "DbVariables",
            "ui.advisory.color": "orange"
        },
        "internal": {
            "interfaces": [
                {
                    "tagged": false,
                    "name": 1.2
                }
            ],
            "class": "VLAN",
            "mtu": 1450
        },
        "external": {
            "interfaces": [
                {
                    "tagged": false,
                    "name": 1.3
                }
            ],
            "class": "VLAN",
            "mtu": 1450
        },
        "failoverAddress": {
            "class": "FailoverUnicast",
            "address": "/Common/HA-self/address"
        },
        "trust": {
            "localUsername": "admin",
            "remoteHost": "192.168.245.111",
            "localPassword": "f5c0nfig",
            "remotePassword": "f5c0nfig",
            "remoteUsername": "admin",
            "class": "DeviceTrust"
        },
        "internal-self": {
            "trafficGroup": "traffic-group-local-only",
            "vlan": "internal",
            "allowService": "default",
            "class": "SelfIp",
            "address": "192.168.40.53/24"
        },
        "class": "Tenant",
        "external-self": {
            "trafficGroup": "traffic-group-local-only",
            "vlan": "external",
            "allowService": "none",
            "class": "SelfIp",
            "address": "192.168.80.58/24"
        }
    }
}
```
Had an error in processing and triggered a rollback.  It generated this error:

```
{"code":500,"message":"java.net.ProtocolException: status:500, body:{\"id\":\"96779d00-43f6-4b89-bd8c-8130aec61f18\",\"selfLink\":\"https://localhost/mgmt/shared/declarative-onboarding/task/96779d00-43f6-4b89-bd8c-8130aec61f18\",\"code\":500,\"status\":\"ERROR\",\"message\":\"rollback failed\",\"errors\":[\"tryUntil: max tries reached: Device group activeStandbyGroup does not exist on this device.\",\"01071413:3: Cannot delete IP (1.1.1.103) because it is used by a system unicast address.\"],\"result\":{\"class\":\"Result\",\"code\":500,\"status\":\"ERROR\",\"message\":\"rollback failed\",\"errors\":[\"tryUntil: max tries reached: Device group activeStandbyGroup does not exist on this device.\",\"01071413:3: Cannot delete IP (1.1.1.103) because it is used by a system unicast address.\"]},\"declaration\":{\"async\":true,\"schemaVersion\":\"1.0.0\",\"label\":\"Cloudinit Onboarding\",\"class\":\"Device\",\"Common\":{\"poolLicense\":{\"licensePool\":\"BIGIPVEREGKEYS\",\"licenseType\":\"licensePool\",\"bigIqHost\":\"192.168.245.114\",\"bigIqUsername\":\"admin\",\"bigIpUsername\":\"admin\",\"reachable\":true,\"class\":\"License\",\"unitOfMeasure\":\"monthly\",\"overwrite\":false},\"ntpServers\":{\"class\":\"NTP\",\"servers\":[\"0.pool.ntp.org\",\"1.pool.ntp.org\",\"2.pool.ntp.org\"]},\"HA-self\":{\"trafficGroup\":\"traffic-group-local-only\",\"vlan\":\"HA\",\"allowService\":\"all\",\"class\":\"SelfIp\",\"address\":\"1.1.1.103/24\"},\"HA\":{\"interfaces\":[{\"tagged\":false,\"name\":\"1.1\"}],\"class\":\"VLAN\",\"mtu\":1450},\"configsync\":{\"configsyncIp\":\"/Common/HA-self/address\",\"class\":\"ConfigSync\"},\"activeStandbyGroup\":{\"saveOnAutoSync\":false,\"autoSync\":true,\"fullLoadOnSync\":false,\"members\":[\"primary-waf-instance.local\",\"secondary-waf-instance.local\"],\"asmSync\":false,\"type\":\"sync-failover\",\"class\":\"DeviceGroup\",\"networkFailover\":false},\"hostname\":\"secondary-waf-instance.local\",\"dnsServers\":{\"nameServers\":[\"8.8.8.8\"],\"search\":[\"example.openstack.com\"],\"class\":\"DNS\"}
```

The FailoverUnicast class is defined in the declaration.. so rollback needs to remove it.

**Comment by focrensh** [comment]
AT-344 for internal tracking

**Comment by jmcalalang** [comment]
can reproduce on "version": "1.4.1",

**Comment by seattlevine** [comment]
Resolved in 1.5.0
